docs/site/DataSources.md

```diff
@@ -81,3 +81,17 @@ Attach the newly created datasource to the app by calling `app.dataSource()`.
 The `app.datasource()` method is available only on application classes
 with `RepositoryMixin` applied.
 " %}
+
```

Comment:
I think the content above (e.g. "Creating a datasource at runtime") should be eventually moved to "How-to guides" and/or the new page "Configuring-datasource.md" you are creating? I guess that's out of scope of this PR, just want to mention it.